### ISSUE TITLE
Timing Controls: New labels and help text for the group of Timing Controls

### ISSUE THREAD

**Issue by terracoda** [phetsims/molecules-and-light#267]
We are adding options for regular speed and slow motion to MAL. We need new state descriptions describing all the timing controls. These changes are based on initial interviews that described the step button and the step interaction as quite confusing.

Updated design and content are also noted in red in the [Control Area section of the design doc](https://docs.google.com/document/d/1r9D3vq4QvmmXkQ5G6HexpybUv_vL6LrYFb3wgPfUdWg/edit#heading=h.z317ugrjbdes) 

@jessegreenberg, I suggest we try to group all the controls together. I'm not sure how the group semantics will play out, but I'd like to try the controls first as a named group.

To get the group semantics, let’s try `aria-labeledby` on a div that surrounds all the timing controls.

**H3: Timing Controls** 
General help text for all the Timing Controls in a paragraph
- "For closer observations choose slow motion, or step forward in time while sim is paused."

**For radio button inputs**, try an un-ordered list. Each input needs the following labels:
- Normal Speed (checked by default)
- Slow Motion 

**Pause/Play, button** 
- Label 1: Pause (when sim is not paused)
- Label 2: Play (when sim is paused)

**Step Forward, button**
Label: Step Forward
Note: disabled when Play button is active.
Extra help text for step button only in a paragraph: 
- "Pause sim to step forward in time."

**Comment by jessegreenberg** [comment]
Sounds good - so if I were to mock up the HTML it may look something like this:
```html
<h3 id='heading-label'>Timing Controls</h3>
<p>For closer oobservations choose slow motion, or step forward in time while sim is paused.</p>
<div aria-labelledby='heading-label'>
  <button>Pause</button>
  <button>Step Forward</button>
  <p>Pause sim to step forward in time</p>
  <ul>
    <li>
      <label id='button1'>Radio Button 1</label>
      <input for='button1' type="radio" checked="checked">
    </li>
    <li>
      <label id="button2"></label>
      <input for='button2' type="radio">
    </li>
  </ul>
</div>
```

And in cases where the slow motion radio buttons are not visible it will look like this:
```html
<h3 id='heading-label'>Timing Controls</h3>
<div aria-labelledby='heading-label'>
  <button>Pause</button>
  <button>Step Forward</button>
  <p>Pause sim to step forward in time</p>
</div>
```

**Comment by terracoda** [comment]
@jessegreenberg, I didn't realize the radiobuttons would disappear or become disabled? Maybe I need to understand this group of controls a bit better. Maybe we need  to more explicitly describe the Play and Pause modes and use more dynamically explicit help text. I've updated the help text in the code examples below.

Also, I was thinking that the speeds offered through the radiobuttons were more useful for the non-visual experience, so I thought they should come first. The default mode is Play.

Based on what you said above, I decided to add in aria-disabled attributes, though I am not 100% sure how to disable radio buttons. I'll need to look into this.

Basically, I was thinking of HTML structure that looks like this when sim is NOT Paused:
```html
<h3 id='heading-label'>Timing Controls</h3>
<p>For closer observations in Play mode, choose slow motion, or Pause sim to step forward in time.</p>
<div aria-labelledby='heading-label'>
 <ul>
    <li>
      <label id='button1'>Normal Speed</label>
      <input for='button1' type="radio" checked="checked" aria-disabled="false">
    </li>
    <li>
      <label id="button2">Slow Motion</label>
      <input for='button2' type="radio" aria-disabled="false">
    </li>
  </ul>

  <button>Pause</button>
  <button aria-disabled="true">Step Forward</button>
  <p>Pause sim to step forward in time.</p>
</div>
```

And if sim IS Paused:
```html
<h3 id='heading-label'>Timing Controls</h3>
<p>For closer observations in Play mode, choose slow motion, or Pause sim to step forward in time.</p>
<div aria-labelledby='heading-label'>
 <ul>
    <li>
      <label id='button1'>Normal Speed</label>
      <input for='button1' type="radio" checked="checked" aria-dsiabled="true">
    </li>
    <li>
      <label id="button2">Slow Motion</label>
      <input for='button2' type="radio" aria-disabled="true">
    </li>
  </ul>

  <button>Play</button>
  <button aria-disabled="false">Step Forward</button>
  <p>Play sim to resume chosen speed.</p>
</div>
```

What do you think? Should we do a simple HTML mock-up first?

**Comment by terracoda** [comment]
Just adding links to aria 1.1 and 1.2:
- https://www.w3.org/TR/wai-aria/#aria-disabled
- draft 1.2: https://rawgit.com/w3c/aria/master/#aria-disabled

I haven't looked at the Authoring Practices yet.

**Comment by jessegreenberg** [comment]
> I didn't realize the radiobuttons would disappear or become disabled?

Yes, they can disappear. Not in molecules-and-light, but I need to instrument the TimeControlNode which is in scenery-phet and supports the case where there are no radio buttons at all (as it was in MAL before this change).

> Also, I was thinking that the speeds offered through the radiobuttons were more useful for the non-visual experience, so I thought they should come first.

We could put them first. But are you OK with them coming first in tab navigation order too? In my opinion these controls are not as important as play/pause buttons, and this order would not be expected for a sighted user.

The rest of your HTML mock ups look great.

**Comment by terracoda** [comment]
I am okay with playing with the order, the Tab order, that is, once we have something implemented. The help text I composed, however, assumes the radiobuttons are coming first in the Tab order. We might have to adjust that as well.

In my mind, I think the Pause/Play button is most important, then the radiobuttons, and then the Step button, but I didn't know how to place the Step button further from the Pause/Play button in the HTML code.

**Comment by terracoda** [comment]
I wonder if a checkbox for slow motion would be better?

**Comment by terracoda** [comment]
@jessegreenberg, I'll mark for discussion at today's meeting.
If it's not too much trouble go ahead implement, but if this could get gnarly, maybe we should discuss with others first?

**Comment by jessegreenberg** [comment]
OK sounds good - I will wait to implement until we discuss further.

**Comment by jessegreenberg** [comment]
We discussed today, notes are in https://docs.google.com/document/d/1r9D3vq4QvmmXkQ5G6HexpybUv_vL6LrYFb3wgPfUdWg/edit#, see "Timing Controls". Before moving forward, I want to discuss with the team one more way to potentially generalize.

**Comment by terracoda** [comment]
At Feb 11 design meeting, we decided to move forward on Timing Controls specifically for this sim. Things might change as work on the general Timing Control node is worked out in scenery issue https://github.com/phetsims/scenery-phet/issues/575 and related issues.

The order, the accessible names and the help text are all in the design doc under the [Timing Controls heading](https://docs.google.com/document/d/1r9D3vq4QvmmXkQ5G6HexpybUv_vL6LrYFb3wgPfUdWg/edit#heading=h.gkri4hptbrio)

Here's kind of what it looks like on start up.

**h3: Timing Controls**
p: For closer observations Play sim on slow speed.

button: {{Pause}}

button: Step Forward, dimmed (disabled)
p: {{Pause sim to step forward little by little.}}

**h4: Sim Speeds**
ul
li: radio button: Normal, selected
li: radio button: Slow

**Comment by terracoda** [comment]
@jessegreenberg, other details are in the design doc. Let me know if you have any questions.

**Comment by terracoda** [comment]
@jessegreenberg, can you use change from "Sim Speeds" to "Play Speeds". I have a feeling that might be more clearly connected to the Play button, and more distinct from the Photon Rate slider's rates.

**Comment by jessegreenberg** [comment]
OK this has been done in the above commits. Implementation follows design comments above and notes in https://docs.google.com/document/d/1r9D3vq4QvmmXkQ5G6HexpybUv_vL6LrYFb3wgPfUdWg/edit#heading=h.gkri4hptbrio.

Closing this issue, review will come with a new dev version.
